Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Managers
+{
+    /// <summary>
+    /// The Mixed Reality manager is responsible for coordinating the operation of the Mixed Reality Toolkit.
+    /// It provides a service registry for all active managers that are used within a project as well as providing the active configuration profile for the project.
+    /// The Profile can be swapped out at any time to meet the needs of your project.
+    /// </summary>
+    public class MixedRealityManager : MonoBehaviour
+    {
+        #region Mixed Reality Manager Profile configuration
+
+        /// <summary>
+        /// The active profile of the Mixed Reality Manager which controls which components are active and their initial configuration.
+        /// *Note configuration is used on project initialization or replacement, changes to properties while it is running has no effect.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The current active configuration for the Mixed Reality project")]
+        private MixedRealityConfigurationProfile activeProfile = null;
+
+        /// <summary>
+        /// The public property of the Active Profile, ensuring events are raised on the change of the configuration
+        /// </summary>
+        public MixedRealityConfigurationProfile ActiveProfile
+        {
+            get { return activeProfile; }
+            set { activeProfile = value; ResetConfiguration(); }
+        }
+
+        private void ResetConfiguration()
+        {
+            foreach (var manager in ActiveProfile.ActiveManagers)
+            {
+                manager.Value.Reset();
+            }
+        }
+
+        #endregion Mixed Reality Manager Profile configuration
+
+        #region Active SDK components
+
+        /// <summary>
+        /// The Active Controllers property lists all the controllers detected by the Mixed Reality manager on startup
+        /// </summary>
+        //[SerializeField]
+        //[Tooltip("The collection of currently active / detected controllers")]
+        //private Controller[] activeControllers = null;
+
+        ///// <summary>
+        ///// The Active Headset property maintains the Headsets/SDK detected by the Mixed Reality manager on startup
+        ///// </summary>
+        //[SerializeField]
+        //[Tooltip("The currently active / detected Headset or SDK")]
+        //private Headset activeHeadset = default(Headset);
```

Comment:
I just commented these out until we use them

Reply:
Fair enough